app/presenters/FacultyPresenter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Michal Struna
+ * Date: 24/11/2018
+ * Time: 16:30
+ */
+
+namespace App\Presenters;
+
+use App\Model\FacultyModel;
+use Nette\Database\DriverException;
+use Nette\Application\UI\Form;
+use Nette\Utils\ArrayHash;
+
+class FacultyPresenter extends BasePresenter {
+
+    private $facultyModel;
+
+    public function __construct(FacultyModel $facultyModel) {
+        $this->facultyModel = $facultyModel;
+    }
+
+    /**
+     * Create edit faculty form.
+     * @return Form Edit faculty form
+     */
+    protected function createComponentEditFacultyForm(): Form {
+        $facultyId = $this->getParameter('id');
+        $faculty = isset($facultyId) ? $this->facultyModel->getById($facultyId) : null;
+
+        $form = new Form;
+        $form->addText('name', 'Název')
+            ->setRequired('Prosím vyplňte název.')
+            ->setHtmlAttribute('autocomplete', 'off')
+            ->setDefaultValue($faculty ? $faculty['nazev'] : '');
+
+        $form->addText('shortName', 'Zkratka')
+            ->setRequired('Prosím vyplňte zkratku.')
+            ->setHtmlAttribute('autocomplete', 'off')
+            ->setDefaultValue($faculty ? $faculty['zkratka'] : '');
+
+        $form->addSubmit('send', $faculty ? 'Upravit' : 'Přidat');
+
+        $form->onSuccess[] = [$this, 'onEdit'];
+        return $form;
+    }
+
+    public function renderDefault(): void {
+        $this->template->faculties = $this->facultyModel->getAll();
+    }
+
+    public function renderEdit(string $id): void {
+
```

Comment:
tady neni nic potreba? :O Chcem tu ty prazdny funkce teda?

Reply:
Když se někde odkazuješ na presenter ve tvaru Faculty:edit, tak to hledá buď renderEdit nebo actionEdit (možná ještě handleEdit?). Pokud ani jedno z toho neexistuje, hází to errory.

Reply:
👍